experimental/dds/tree/src/SharedTree.ts

```diff
@@ -433,14 +433,14 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 		}
 	}
 
-	private idCompressor: IdCompressor;
+	private _idCompressor: IdCompressor;
```

Comment:
this pattern scares me, as it would be easy to accidentally use the wrong one. perhaps we could shadow it with a private getter and comment it appropriately?

Reply:
wait, is it a property on runtime? if so, why rename?

Reply:
Because `idCompressor` is now a property on `SharedObject`, anywhere in `SharedTree` where we call `this.idCompressor` would now be using the runtime compressor. I'm not a fan of this rename either, but I'm not sure what we want to do here

Reply:
If you do the shadowing as Taylor suggested, then you can still make the original available, e.g. in the constructor, before you do the shadowing, bind a new property `this.runtimeIdCompressor` to `this.idCompressor`. It's not perfect but at least someone will think twice before using the wrong one, especially if it's documented.

Or, if we're only going to ever use one or the other, just do the shadowing without the extra property.

Reply:
I don't think we should *ever* use the runtime one in that class.

Reply:
This is a little trickier to fix than it looks at first glance. Because experimental SharedTree use the old, non-forked compressor with attribution, the types no longer align. To my knowledge, I'm unable to just re-assign the property because the types are now incompatible. Is there a good workaround for this?

Reply:
Bummer. Well, at the very least, shadow the SharedObject `idCompressor` with a getter that immediately errors at runtime if invoked, and do all the docs.

Reply:
Okay, nice. I can't think of anything better, so that's good enough for me here. Not sure if @taylorsw04 has had any more ideas since he started this thread.